### PR TITLE
Scroll to top of the page when changing story

### PR DESCRIPTION
**Pull request by Hypnosphi**
Issue: #2729

### CODE REVIEW

File: lib/core/src/client/preview/start.js [hunk 1]
```diff
@@ -52,10 +52,20 @@ export default function start(render, { decorateStory } = {}) {
...
 
+  let prevKind;
+  let prevStory;
+
   // initialize the UI
   const renderUI = () => {
     if (isBrowser) {
       render(context);
+      const { selectedKind, selectedStory } = reduxStore.getState();
+      if (selectedKind !== prevKind || selectedStory !== prevStory) {
```
Comment (tmeasday) [inline lib/core/src/client/preview/start.js]: Perhaps we shouldn't call into `render()` at all it if they are the same. I'm pretty sure all frameworks end up making this same check (although Angular ends up re-rendering, it is not clear why)

**Review by tmeasday — APPROVED** [review]
Happy to approve this, what do you think about the suggestion? Could be a separate PR

File: lib/core/src/client/preview/start.js (hunk 1 above)
Comment (Hypnosphi) [reply lib/core/src/client/preview/start.js]: Makes sense
@igor-dv @alterx WDYT?

Comment (Hypnosphi) [reply lib/core/src/client/preview/start.js]: There can be also `state.error`

Comment (tmeasday) [reply lib/core/src/client/preview/start.js]: Yes, good point. This is something I've wanted to refactor out anyway, I will get it in a separate PR.

**Pull request merged**